internal/gps/project_root_cache.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gps
+
+import "github.com/pkg/errors"
+
+// ProjectRootCache caches the result of deducing a project root from a
+// source manager, which is quite helpful when repeatedly converting from
+// import paths to root projects.
+type ProjectRootCache struct {
+	prs map[string]ProjectRoot
```

Comment:
It didn't seem to me that it was caching the result in the source manager?

The source manager calls out to the deducer: 
https://github.com/golang/dep/blob/7d365252964cbb738685d2ed2ae26f208e111424/internal/gps/deduce.go#L577

Then the deducer figures it out fresh each time:
https://github.com/golang/dep/blob/7d365252964cbb738685d2ed2ae26f208e111424/internal/gps/deduce.go#L577